test/aerospike_clj/client_test.clj

```diff
@@ -69,14 +69,74 @@
     (testing "clojure maps can be serialized as-is")
     (let [v @(client/get-single-no-meta *c* K _set)]
       (is (= data v)) ;; per value it is identical
-      (is (= java.util.HashMap (type v))) ;; but we get back a HashMap
-      (is (false? (map? v)))
+      (is (= clojure.lang.PersistentArrayMap (type v)))
       (testing "using Jackson to recursively create a PersistantHashMap"
         (let [json (.writeValueAsString (ObjectMapper.) v)
               parsed (json/parse-string json #(keyword (subs % 1)))]
           (is (= data parsed))
           (is (true? (map? parsed))))))))
 
+(deftest put-multiple-bins-get-clj-map
+  (let [data {:foo {:bar [(rand-int 1000)]}
+              :baz true
+              :qux false
+              :quux nil}]
+    (is (true? @(client/create *c* K _set data 100)))
+    (testing "clojure maps can be serialized from bins")
+    (let [v @(client/get-single-no-meta *c* K _set)]
+      (is (= (:foo data) (:foo v))) ;; per value it is identical
+      (is (= (:baz data) (:baz v))) ;; true value returns the same after being sanitized/desanitized
+      (is (= (:qux data) (:qux v))) ;; false value returns the same after being sanitized/desanitized
+      (is (= (:quux data) (:quux v))) ;; nil value retuns the same after being sanitized/desanitized
+      (is (= clojure.lang.PersistentArrayMap (type v))) ;; converted back to a Clojure map instead of HashMap
+      (is (true? (map? v)))
+      (testing "using Jackson to recursively create a PersistantHashMap"
```

Comment:
this piece of code was part of the tests already because users accidentally put maps in the single-bin versions API, and those were serialized by Java to `HashMap`s. I added it in order to prove that there is a technical way of easily recursively convert it into a Clojure map. So no need for it here.

Reply:
removed